test/proxy_tests.cpp

```diff
@@ -86,6 +86,25 @@ TEST(ProxyTests, ReferenceProxySessionTest) {
     EXPECT_EQ(ErrorCode::OK, response.error.code);
 }
 
+if 0
+TEST(ProxyTests, NoProxyTest) {
```

Comment:
Please properly test your changes and let me know how I can verify them.

Reply:
Added a test.
The test needs direct connection to the web + free proxy.